code/game/objects/structures/artstuff.dm

```diff
@@ -16,12 +16,17 @@
 //Adding canvases
 /obj/structure/easel/attackby(obj/item/I, mob/user, params)
 	if(istype(I, /obj/item/canvas))
-		var/obj/item/canvas/C = I
-		user.drop_item(C)
-		painting = C
-		C.forceMove(get_turf(src))
-		C.layer = layer+0.1
-		user.visible_message("<span class='notice'>[user] puts \the [C] on \the [src].</span>","<span class='notice'>You place \the [C] on \the [src].</span>")
+		if(!painting)
+			var/obj/item/canvas/C = I
+			user.drop_item(C)
+			painting = C
+			C.forceMove(get_turf(src))
+			C.layer = layer+0.1
+			C.pixel_x = initial(C.pixel_x)
+			C.pixel_y = initial(C.pixel_y)
```

Comment:
Because I'm unfamiliar with canvases, what happens if you don't have these two lines?

Reply:
Because if you place canvases on tables and pick them up, the offsets end up stacking because of the pixel placement system of tables

Reply:
Makes sense :+1: